pkg/v1/remote/transport/transport.go

```diff
@@ -101,3 +101,14 @@ func NewWithContext(ctx context.Context, reg name.Registry, auth authn.Authentic
 		return nil, fmt.Errorf("unrecognized challenge: %s", pr.challenge)
 	}
 }
+
+// Transport results in *not* wrapping supplied transport with additional logic such as retries, useragent and debug logging
+// Consumers are opt-ing into providing their own transport without any additional wrapping.
+type Transport struct {
```

Comment:
I think all that's left is to make the `transport` package actually return these. 

I don't love the `transport.Transport` name. Maybe `transport.Wrapper` or something?

Another issue is that it's kind of a pain to actually construct a transport outside of the remote package (you need to know where layers came from, where you're pushing, etc). We could make that a lot more ergonomic by introducing a new transport implementation that defers those decisions until they're needed, like containerd (https://github.com/google/go-containerregistry/issues/666#issuecomment-580034491).

Reply:
Yep! changed it to `transport.Wrapper`

>  make the transport package actually return these.

I wasn't 100% sure what you meant here. I took this to mean `NewWithContext` should return `transport.Wrapper`. I pushed the change.

> Another issue is that it's kind of a pain to actually construct a transport outside of the remote package (you need to know where layers came from, where you're pushing, etc). We could make that a lot more ergonomic by introducing a new transport implementation that defers those decisions until they're needed, like containerd (#666 (comment)).

hmm ok, is it possible to split introducing a new deferring transport implementation into a separate PR? Or do you see this as a blocker to this PR?

Reply:
> I wasn't 100% sure what you meant here. I took this to mean NewWithContext should return transport.Wrapper. I pushed the change.

Yep, that's perfect.

> hmm ok, is it possible to split introducing a new deferring transport implementation into a separate PR? Or do you see this as a blocker to this PR?

Not a blocker, just thinking out loud.